api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -181,6 +181,32 @@
         return getOptionalValue(propertyName, arrayType).map(Arrays::asList);
     }
 
+    /**
+     * Return a {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} 
+     * instance for the specified type.
+     * @param <T> 
+     *              The {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} Class Type
+     * @param configProperties
+     *              The class of the {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties}
+     * @param prefix
+     *              The prefix for the config properties included in the class specified by configProperties
+     * @return      An instance of the {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} 
+     * for the specified type and prefix
+     */
+    <T> T getConfigProperties(Class<T> configProperties, String prefix);
+
+        /**
+     * Return a {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} 
+     * instance for the specified type.
+     * @param <T> 
+     *              The {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} Class Type
+     * @param configProperties
+     *              The class of the {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties}
+     * @return      An instance of the {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties configuration properties} 
+     * for the specified type
+     */
+    <T> T getConfigProperties(Class<T> configProperties);
+
```

Comment:
I find the javadoc of both methods to generic, maybe even misleading. It does not mention that the class passed does not have to have the `ConfigProperties` annotation. I almost gives the impression the annotation is required. Also I would like to see a description what prefix will be use in what case, as far as I remember these should be:

* class without annotation => empty/no prefix
* class with annotation => prefix from annotation
* prefix provided by parameter => always use that prefix, ignore prefix from annotation if exists

Reply:
@jbee the spec clarifies and prefix order. Quote below

* programmatic look up via `Config.getConfigProperties()`
[source, java]
----
Config config = ConfigProvider.getConfig();
Details serverDetails = config.getConfigProperties(Details.class);
Details clientDetails = config.getConfigProperties(Details.class, "client"); 
Details details = config.getConfigProperties(Details.class, ""); //use no prefix
----

In the above example, the `serverDetails` will contain the following info, because the prefix `server` on the bean class `Details` is in action:

```
serverDetails.host -> server.host -> localhost
serverDetails.port -> server.port -> 9080
serverDetails.endpoint -> server.endpoint -> query
serverDetails.getLocation() -> server.old.location -> London
```
The `clientDetails` will retrieve the following properties, because the specified prefix `client` overrides the prefix on the bean class.

```
clientDetails.host -> client.host -> myHost
clientDetails.port -> client.port -> 9081
clientDetails.endpoint -> client.endpoint -> shelf
clientDetails.getLocation() -> client.old.location -> Dublin
```
The `empty` prefix overrides the prefix set on the bean class `server`. Therefore, this will retrieve the following properties.

```
details.host -> host -> anotherHost
details.port -> port -> 9082
details.endpoint -> endpoint -> book
details.getLocation() -> old.location -> Berlin
```
I also added some tck for this.

I'll mention the `@ConfigProperties` annotation does not need to be present for programmatic lookup and then improve the javadoc.

Reply:
I have updated the javadoc @jbee! Please take a look to see whether you have further comments!